### PR TITLE
UCP/TAG/TEST: Create separate structure for tag-matching queues.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1396/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3325/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1397/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1398/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1399/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1400/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3326/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@alex-mikheev @brminich @shamisp pls review

File: src/ucp/tag/tag_match.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2015.  ALL RIGHTS RESERVED.
```
Comment (brminich) [inline src/ucp/tag/tag_match.h]: minor: 2017 can be specified

Comment (brminich) [inline src/ucp/tag/tag_match.h]: :+1:

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1407/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3332/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3336/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3338/ for details (Mellanox internal link).

**Pull request merged**